src/ekss/adapters/outbound/vault/client.py

```diff
@@ -32,19 +33,39 @@ def __init__(self, config: VaultConfig):
         self._client = hvac.Client(url=config.vault_url, verify=config.vault_verify)
         self._path = config.vault_path
 
-        self._role_id = config.vault_role_id.get_secret_value()
-        self._secret_id = config.vault_secret_id.get_secret_value()
+        self._kube_role = config.vault_kube_role
+        if self._kube_role:
+            # use kube role and service account token
+            self._kube_role = self._kube_role
+            self._kube_adapter = Kubernetes(self._client.adapter)
+            self._service_account_token_path = config.service_account_token_path
+        elif config.vault_role_id and config.vault_secret_id:
```

Comment:
Got an error here that `None` doesn't support the `get_secret_value()` method.

Reply:
Are you sure? I do not get that error (with mypy 1.7.0  as per the log file). The `elif` statements makes sure that `vault_role_id` is not faulty, so it cannot be None, and mypy should understand that.

Reply:
Sorry, I worded that a bit clumsily, I meant before I added the conditional

Reply:
Ah ok, did not see that you added the `elif`. It was an `else` before, right?

The `elif` is the proper fix, but you should then also add an `else` block afterwards, which would then catch the case that is currently handled by the `if not all...` statement, simply raising an error.

Reply:
I.e. the `not all...` statement can be removed (is already covered by the `elif`), and the `raise` statement can go to the added `else` block.

Reply:
I guess this change then also needs to be backported to FIS.